api/src/main/java/brooklyn/mementos/BrooklynMementoPersister.java

```diff
@@ -97,10 +97,13 @@
     /** @deprecated since 0.7.0, use {@link #checkpoint(BrooklynMementoRawData, PersistenceExceptionHandler)} 
      * and javadoc on implementations of that */ @Deprecated  // pretty sure this is not used outwith deprecated code
     void checkpoint(BrooklynMemento memento, PersistenceExceptionHandler exceptionHandler);
-    
-    void checkpoint(BrooklynMementoRawData newMemento, PersistenceExceptionHandler exceptionHandler);
 
+    /** applies a full checkpoint (write) of all state */  
+    void checkpoint(BrooklynMementoRawData newMemento, PersistenceExceptionHandler exceptionHandler);
+    /** applies a partial write of state delta */  
     void delta(Delta delta, PersistenceExceptionHandler exceptionHandler);
+    /** inserts an additional delta to be written on the next delta request */
+    void queueDelta(Delta delta);
```

Comment:
The semantics of `queueDelta` (to only be applied on the next delta request) is clear and good enough for current usage, but feels strange to have on an api. If means your queue won't be executed until either you call delta (in which case why did you queue it) or someone else calls delta (in which case you don't know when that will happen).

Worth at least having `@Beta` on this.